gcore/gdaldrivermanager.cpp

```diff
@@ -508,6 +524,43 @@ int GDALDriverManager::RegisterDriver(GDALDriver *poDriver)
     if (poDriver->pfnVectorTranslateFrom != nullptr)
         poDriver->SetMetadataItem(GDAL_DCAP_VECTOR_TRANSLATE_FROM, "YES");
 
+    if (m_bInDeferedDriverLoading)
+    {
+        if (m_oMapRealDrivers.find(poDriver->GetDescription()) !=
+            m_oMapRealDrivers.end())
+        {
+            CPLError(
+                CE_Failure, CPLE_AppDefined,
+                "RegisterDriver() in m_bInDeferedDriverLoading: %s already "
+                "registered!",
+                poDriver->GetDescription());
+            delete poDriver;
```

Comment:
just wondering if poDriver could be automatically managed with a unique_ptr (and use std::move later).

Reply:
That would be good indeed, but GDALDriverManager::RegisterDriver() is an existing public method used by all drivers. Changing its signature would affect all existing in-tree and out-of-tree driver. So I'm leaving that out